zerocopy-derive/src/lib.rs

```diff
@@ -528,6 +529,45 @@ fn derive_unaligned_inner(ast: &DeriveInput, _top_level: Trait) -> Result<TokenS
     }
 }
 
+fn derive_hash_inner(ast: &DeriveInput, _top_level: Trait) -> Result<TokenStream, Error> {
+    let type_ident = &ast.ident;
+    let (impl_generics, ty_generics, where_clause) = ast.generics.split_for_impl();
+    let where_predicates = where_clause.map(|clause| &clause.predicates);
+    Ok(quote! {
```

Comment:
One last (I think) nit: Could you leave a comment why deferring to `impl_block` isn't appropriate here?

Reply:
Done!